src/test/java/springboot/service/BookServiceTest.java

```diff
@@ -0,0 +1,237 @@
+package springboot.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import springboot.dto.book.BookDto;
+import springboot.dto.book.BookDtoWithoutCategoryIds;
+import springboot.dto.book.CreateBookRequestDto;
+import springboot.exeptions.EntityNotFoundException;
+import springboot.mapper.BookMapper;
+import springboot.model.Book;
+import springboot.model.Category;
+import springboot.repository.BookRepository;
+import springboot.repository.CategoryRepository;
+import springboot.service.impl.BookServiceImpl;
+import springboot.testutil.BookUtil;
+import springboot.testutil.CategoryUtil;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private CategoryRepository categoryRepository;
+    @Mock
+    private BookMapper bookMapper;
+
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    private Book book;
+    private Category category;
+    private BookDto bookDto;
+    private CreateBookRequestDto requestDto;
+    private Pageable pageable;
+    private Page<Book> bookPage;
+
+    @BeforeEach
+    public void setUp() {
+        requestDto = BookUtil.createBookRequestDto();
+        book = BookUtil.getBook(requestDto);
+        category = CategoryUtil.createCategory();
+        bookDto = BookUtil.createBookDto(book);
+        pageable = BookUtil.createPageable();
+        bookPage = BookUtil.createBookPage(pageable);
+    }
+
+    @DisplayName("Create a book and return BookDto")
+    @Test
+    public void createBook_WithValidData_ReturnsBookDto() {
+        when(bookMapper.toModel(requestDto)).thenReturn(book);
+        when(bookRepository.save(book)).thenReturn(book);
+        when(bookMapper.toDto(book)).thenReturn(bookDto);
+        when(categoryRepository.findById(101L)).thenReturn(Optional.of(category));
+        BookDto saveBook = bookService.createBook(requestDto);
+
+        assertThat(saveBook).isEqualTo(bookDto);
+        assertThat(saveBook).isNotNull();
+        verify(bookRepository, times(1)).save(book);
```

Comment:
times(1) by default - remove (check other places)
```suggestion
        verify(bookRepository).save(book);
```